lib/index.js

```diff
@@ -76,6 +78,37 @@ class PercySeleniumClient {
   async finalizeBuild() {
     await this.percyClient.finalizeBuild(this.buildId);
   }
+
+  async domSnapshot() {
+    const docType = await this.getDoctype();
+    const dom = await this.driver.executeScript(
+      // eslint-disable-next-line
+      () => window.document.documentElement.outerHTML,
+    );
+    return docType + dom;
+  }
+
+  async getDoctype() {
+    const doctype = await this.driver.executeScript(
+      // eslint-disable-next-line
+      () => window.document.doctype,
+    );
+
+    return doctype ? this.doctypeToString(doctype) : DEFAULT_DOCTYPE;
+  }
+
+  doctypeToString(doctype) {
+    const publicDeclaration = doctype.publicId
+      ? ` PUBLIC "${doctype.publicId}" `
+      : '';
+    const systemDeclaration = doctype.systemId
+      ? ` SYSTEM "${doctype.systemId}" `
+      : '';
+
+    return `<!DOCTYPE ${
```

Comment:
wouldn't this add a second docType if it's already present? guess not going to be a problem

Reply:
See https://github.com/4Catalyzer/percy-seleniumjs/blob/e158f21fb1bc54b51fece3786907cf021b9b255f/lib/index.js#L88. I get the outerHTML and doctype and concatenate

Reply:
Tested by running my test_integration circleci job locally and percy outputted the correct HTML

Reply:
👍